gateway/src/resty/synchronization.lua

```diff
@@ -53,4 +53,28 @@ function _M:release(key)
   semaphores[key] = nil
 end
 
+--- run a new function, callback using locks on the given key
+-- @tparam string key: key for the semaphore
+-- @tparam int timeout: timeout for getting the lock before raise an error.
+-- @tparam function callback: function to execute if the lock is adquired correctly
```

Comment:
Maybe we should also add `...` to the documentation of the function. I can't remember the syntax now.